pinot-core/src/main/java/org/apache/pinot/core/data/recordtransformer/NullValueTransformer.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.data.recordtransformer;
+
+import java.util.Collection;
+import javax.annotation.Nullable;
+import org.apache.pinot.common.data.FieldSpec;
+import org.apache.pinot.common.data.Schema;
+import org.apache.pinot.core.data.GenericRow;
+
+
+public class NullValueTransformer implements RecordTransformer {
+
+  private final Collection<FieldSpec> _fieldSpecs;
+  private Schema _schema;
+
+  public NullValueTransformer(Schema schema) {
+    _schema = schema;
+    _fieldSpecs = _schema.getAllFieldSpecs();
+  }
+
+  @Nullable
+  @Override
+  public GenericRow transform(GenericRow row) {
+    for (FieldSpec fieldSpec : _fieldSpecs) {
+      String fieldName = fieldSpec.getName();
+      if (row.getValue(fieldName) == null && fieldSpec.getFieldType() != FieldSpec.FieldType.TIME) {
```

Comment:
Make the `fieldSpec.getFieldType() != FieldSpec.FieldType.TIME` as a precondition?

Reply:
Why should it be a precondition?

Reply:
Time column should never have null value. Passing null as time value will cause exception later

Reply:
Not sure if its the responsibility of the transformer to verify that. We can't do much in real-time case if the time column is null i.e. throwing Exception will not help.

Reply:
We assume no null value in the following transformers, so this transformer should guarantee that, or throw exception. Performing a check and throw exception here is more clear than getting NPE somewhere in the following transformers, and easier to debug.

Reply:
what will happen in the real-time case when an exception is thrown

Reply:
Actually TimeTransformer can handle null values correctly, and throw exception if both incoming and outgoing values are null, so we are good here.